Copy style from labels to bundles.

ahelsley opened the pull request:
Fixes #12 
Addresses #13

teresa-ou commented:
Thanks for putting together this PR! 

Overall, the code looks good to me, but my main concerns are about the resulting UI:

* The color choices that Gmail provides for the labels, don't always work well as the background color for a bundle row. I think this is because we're taking the background color from a small component and applying it to a much larger component. For example, the reds can look alarming, or the blues can be confused with the blue styling that's shown when an email is selected with the checkbox. Since not all colors look good, this places a burden on the user to choose suitable colors.
* When the bundle is expanded, the bundle row and a div with class `bundle-area` together form a box that is behind the messages in the bundle. In this changeset, the bundle-area is still grey, while the bundle row is colored.
* I think I would prefer a less-is-more approach to color, such as the suggestion of coloring the bundle icon in #12 (not sure if the colors work well against the grey background of the emails) or a [label-shaped indicator](https://www.ghacks.net/2018/09/13/google-terminates-google-inbox-service/) of the number of new messages.

ahelsley commented:
Agreed, the styling is probably too aggressive.  Perhaps copying salient aspects of the label-style to a more-restricted region around the bundle-name would mimic Gmail's existing Label styling without being a detriment to the aesthetics and usability of inboxy.

I've updated my master to reflect a more subtle color scheme:

* Added subtle coloring to icon background, influenced by label color.
* Limited opaque Gmail-label styling solely to the area around bundle name.
* Applied subtle label-colored gradient to BundleRow.
* Updated bundle-area to blend with the color found in the BundleRow using a linear gradient starting with a transparent color.

teresa-ou commented:
The overall look and feel are still a significant departure from Gmail's original styles. I think we should stay away from using gradients, since Gmail mainly uses flat areas of color. As for the colored boxes around the bundle names, these boxes look similar to labels, which makes it seem like a secondary component rather than the main text for those rows. 

Since we're mainly discussing UI, perhaps we should iterate over it on the Github issue instead of in a PR. Also, it might be quicker to make UI mocks in design software (ex. Figma) than to code up designs, but feel free to use whichever strategy you're most comfortable with.

atippett commented:
I love the pull to keep the UI simple and clean and also think some separation of the bundles by their label color would be nice.  The bundles don't really stand out enough and I find myself searching for them.  I honestly forget what inbox felt like 🤷 

![Screen Shot 2020-08-06 at 8 26 11 AM](https://user-images.githubusercontent.com/189563/89550593-a6c66780-d7be-11ea-89a9-9f0145c5e7db.png)

tmenier commented:
@teresa-ou is merging this PR (or otherwise addressing #12) still under consideration? Incorporating the label colors into the bundled view seems to make perfect sense logically, and it's hard to imagine any other logged enhancement making a larger impact visually. In other words, it seems like a very "good" idea. :)

lpakula commented:
Are there any plans to move this feature forward? It would be a very nice addition

JoeriTheGreat commented:
@lpakula added this to his fork. https://github.com/lpakula/inboxy
I tried his fork and the bundle colors where working perfectly
